File: solutions/julia/grains/1/grains.jl
"""Calculate the number of grains on square `square`."""
function on_square(square)
    square <= 0 || square > 64 && throw(DomainError(square))
    big(2)^(square-1)
end

"""Calculate the total number of grains after square `square`."""
function total_after(square)
    (square <= 0 || square > 64) && throw(DomainError(square))
    ans = big(0)
    p = big(1)
    for _ in 1:square
        ans += p
        p *= big(2)
    end
    ans
end

File: solutions/csharp/grains/1/Grains.cs
public static class Grains
{
    public static ulong Square(int n)
    {
        if (n <= 0 || n > 64)
        {
            throw new ArgumentOutOfRangeException();
        }
        else
        {
            return (ulong)Math.Pow(2, n - 1);
        }
    }

    public static ulong Total()
    {
        var ans = (ulong)0;
        var p = (ulong)1;
        for (var i = 0; i < 64; i++)
        {
            ans += p;
            p *= (ulong)2;
        }
        return ans;
    }
}